libs/sysaudio/src/coreaudio.zig

```diff
@@ -138,28 +138,32 @@ pub const Context = struct {
                 }
             }
 
-            prop_address.mSelector = if (is_darling) c.kAudioDevicePropertyPreferredChannelsForStereo else c.kAudioDevicePropertyPreferredChannelLayout;
-            prop_address.mScope = if (mode == .playback) c.kAudioObjectPropertyScopeOutput else c.kAudioObjectPropertyScopeInput;
-            if (c.AudioObjectGetPropertyDataSize(id, &prop_address, 0, null, &io_size) != 0) {
+            // for now, only set `channels` to the output channels
+            const audio_buffer_list_property_address = c.AudioObjectPropertyAddress{
+                .mSelector = c.kAudioDevicePropertyStreamConfiguration,
+                .mScope = c.kAudioDevicePropertyScopeOutput,
+                .mElement = c.kAudioObjectPropertyElementMain,
+            };
+            var output_audio_buffer_list: c.AudioBufferList = undefined;
+            var audio_buffer_list_size: c_uint = undefined;
+
+            if (c.AudioObjectGetPropertyDataSize(id, &audio_buffer_list_property_address, 0, null, &audio_buffer_list_size) != c.noErr) {
                 return error.OpeningDevice;
+                // return error.FailedToGetAudioBufferListSize;
             }
 
-            var channel_layout: c.AudioChannelLayout = undefined;
-            if (c.AudioObjectGetPropertyData(
-                id,
-                &prop_address,
-                0,
-                null,
-                &io_size,
-                &channel_layout,
-            ) != 0) {
+            if (c.AudioObjectGetPropertyData(id, &prop_address, 0, null, &audio_buffer_list_size, @ptrCast(*anyopaque, &output_audio_buffer_list)) != c.noErr) {
                 return error.OpeningDevice;
+                // return error.FailedToGetAudioBufferList;
```

Comment:
?

Reply:
I was writing the code backwards and didn't realize this specific function needed to adhere to an abstract set. I left this in because these errors are more informative as to what failed. Is there a reason why the refresh error set can't be inferred? (Can you swap backends during program execution?)

Reply:
error sets are [defined](https://github.com/hexops/mach/blob/81bcce0c482b0a470282c95840e0561585a4adc9/libs/sysaudio/src/main.zig#L61-L65) and you can only change backend at initialization (runtime code).

Reply:
Sounds good :)